src/image-tvg.c

```diff
@@ -0,0 +1,1453 @@
+    case TVG_STYLE_RADIAL:
+        /* TODO: Implement radial gradient color */
+        twin_paint_stroke(ctx->pixmap,
+                          GET_COLOR(ctx, fill_style->radial.color0), ctx->path,
+                          pen_width);
+        break;
+    }
+}
+
+static void _fill_path_with_style(tvg_context_t *ctx,
+                                  const tvg_style_t *fill_style)
+{
+    switch (fill_style->kind) {
+    case TVG_STYLE_FLAT:
+        twin_paint_path(ctx->pixmap, GET_COLOR(ctx, fill_style->flat),
+                        ctx->path);
+        break;
+    case TVG_STYLE_LINEAR:
+        /* TODO: Implement linear gradient color */
+        twin_paint_path(ctx->pixmap, GET_COLOR(ctx, fill_style->linear.color0),
+                        ctx->path);
+        break;
+    case TVG_STYLE_RADIAL:
+        /* TODO: Implement radial gradient color */
+        twin_paint_path(ctx->pixmap, GET_COLOR(ctx, fill_style->radial.color0),
+                        ctx->path);
+        break;
+    }
+}
+
+static tvg_result_t tvg_parse_fill_rectangles(tvg_context_t *ctx,
+                                              size_t size,
+                                              const tvg_style_t *fill_style)
+{
+    size_t count = size;
+    tvg_result_t res;
+    tvg_rect_t r;
+    twin_path_t *path = ctx->path;
+    while (count--) {
+        res = tvg_parse_rect(ctx, &r);
+        if (res != TVG_SUCCESS)
+            return res;
+        twin_path_rectangle(path, D(r.x), D(r.y), D(r.width), D(r.height));
+        _fill_path_with_style(ctx, fill_style);
+        twin_path_empty(path);
+    }
+    return TVG_SUCCESS;
+}
+
+static tvg_result_t tvg_parse_line_fill_rectangles(
+    tvg_context_t *ctx,
+    size_t size,
+    const tvg_style_t *fill_style,
+    const tvg_style_t *line_style,
+    float line_width)
+{
+    size_t count = size;
+    tvg_result_t res;
+    tvg_rect_t r;
+    if (line_width == 0) {
+        line_width = .01;
+    }
+    twin_path_t *path = ctx->path;
+    while (count--) {
+        res = tvg_parse_rect(ctx, &r);
+        if (res != TVG_SUCCESS)
+            return res;
+
+        twin_path_rectangle(path, D(r.x), D(r.y), D(r.width), D(r.height));
+        _fill_path_with_style(ctx, fill_style);
+        _stroke_path_with_style(ctx, line_style, D(line_width));
+        twin_path_empty(path);
+    }
+    return TVG_SUCCESS;
+}
+
+static tvg_result_t tvg_parse_fill_paths(tvg_context_t *ctx,
+                                         size_t size,
+                                         const tvg_style_t *style)
+{
+    tvg_result_t res = TVG_SUCCESS;
+    uint32_t *sizes = (uint32_t *) malloc(size * sizeof(uint32_t));
+    if (sizes == NULL) {
+        return TVG_E_OUT_OF_MEMORY;
+    }
+    for (size_t i = 0; i < size; ++i) {
+        res = tvg_read_varuint(ctx, &sizes[i]);
+        ++sizes[i];
+        if (res != TVG_SUCCESS) {
+            goto error;
+        }
+    }
+    twin_path_t *path = ctx->path;
+    if (!path) {
```

Comment:
It checks whether the `path` is NULL here. However, on line 946, it does not perform this check.

Reply:
I think this checking could be remove since we check path is NULL initially.